clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/StoreIngestionTask.java

```diff
@@ -327,6 +328,10 @@ public abstract class StoreIngestionTask implements Runnable, Closeable {
   private final Runnable runnableForKillIngestionTasksForNonCurrentVersions;
   protected final AtomicBoolean recordLevelMetricEnabled;
 
+  private final StoreVersionRoleChangedListener storeVersionRoleChangedListener;
+
+  protected TopicPartitionReplicaRole.VersionRole versionRole;
```

Comment:
What is its initial value?
The very first subscription action triggered by Helix is using this value as SOT.

Reply:
The initial value of this versionRole is set by `storeVersionRoleChangedListener` per store version initially in the constructor. If we decided to move `storeVersionRoleChangedListener` into `KafkaIngestionService`, I think we may need to add logic to decide the role initially.